php/class-wp-seo-social-settings.php

```diff
@@ -0,0 +1,708 @@
+<?php
+/**
+ * Class file for WP_SEO_Social_Settings
+ *
+ * @package WP_SEO_Social
+ */
+
+/**
+ * Manages extension of WP_SEO_Settings
+ */
+class WP_SEO_Social_Settings {
+	/**
+	 * Instance of this class.
+	 *
+	 * @var object
+	 */
+	private static $instance = null;
+
+	/**
+	 * Post types that can be viewed individually and have per-entry meta values.
+	 *
+	 * @see WP_SEO_Social_Settings::setup().
+	 *
+	 * @var array Post type objects.
+	 */
+	private $single_post_types = array();
+
+	/**
+	 * Post types with archives, which can have meta fields set for them.
+	 *
+	 * @see  WP_SEO_Social_Settings::setup().
+	 *
+	 * @var array Post type objects.
+	 */
+	private $archived_post_types = array();
+
+	/**
+	 * Taxonomies with archive pages, which can have meta fields set for them.
+	 *
+	 * @see  WP_SEO_Social_Settings::setup().
+	 *
+	 * @var array Term objects.
+	 */
+	private $taxonomies = array();
+
+	/**
+	 * Fields that are classified as text fields
+	 *
+	 * @see  WP_SEO_Social_Settings::register_settings().
+	 *
+	 * @var array Field ID's.
+	 */
+	private $handle_as_text = array();
+
+	/**
+	 * Fields to whitelist.
+	 *
+	 * @var array Field ID's to whitelist.
+	 */
+	public $fields_to_whitelist = array(
+		'og_title',
+		'og_description',
+		'og_image',
+		'og_type',
+	);
+
+	/**
+	 * The default options to save.
+	 *
+	 * @var array.
+	 */
+	public $default_options = array();
+
+	/**
+	 * Storage unit for the current option values of the plugin.
+	 *
+	 * @var array.
+	 */
+	public $options = array();
+
+	/**
+	 * Get the instance of this class.
+	 */
+	public static function instance() {
+		if ( ! isset( self::$instance ) ) {
+			self::$instance = new WP_SEO_Social_Settings;
+			self::$instance->setup();
+		}
+		return self::$instance;
+	}
+	/**
+	 * Setup the class
+	 */
+	protected function setup() {
+		add_action( 'admin_init', array( $this, 'set_properties' ), 9 );
+		if ( is_admin() ) {
+			add_action( 'admin_init', array( $this, 'register_settings' ) );
+		}
+		add_filter( 'wp_seo_sanitize', array( $this, 'sanitize' ), 10, 2 );
+		add_filter( 'wp_seo_options_page_menu_title', function() {
```

Comment:
In my opinion I think it is easier to read the code when each function is is declared at the class level with documentation.  If you are going to add anonymous functions as the callbacks then I think adding docblocks will help with the readability of the code :)